src/pages/Import.js

```diff
@@ -0,0 +1,186 @@
+      spinner
+    )
+  }
+
+  componentDidMount() {
+    const { path, version } = this.props
+
+    Dockstore.getWdl(path, version).then(
+      ({ descriptor }) => this.setState({ wdl: descriptor }),
+      failure => this.setState({ loadError: failure })
+    )
+
+    Rawls.workspacesList().then(workspaces => this.setState({ workspaces }))
+  }
+
+  renderImport = () => {
+    function section(title, flex, contents) {
+      return div(
+        { style: { flex, overflow: 'hidden', margin: '3rem' } },
+        [
+          div({ style: _.merge({ fontWeight: 500, marginBottom: '1rem' }, Style.elements.sectionHeader) }, title),
+          contents
+        ])
+    }
+
+    return div({ style: { display: 'flex' } },
+      [
+        section('Importing', 5, this.renderWdlArea()),
+        section('Destination Project', 3, this.renderWorkspaceArea())
+      ]
+    )
+  }
+
+  renderWdlArea = () => {
+    const { path, version } = this.props
+    const { wdl } = this.state
+
+    return div(
+      {
+        style: {
+          borderRadius: 5, backgroundColor: 'white', padding: '1rem',
+          boxShadow: `0 0 2px 0 rgba(0,0,0,0.12), ${Style.standardShadow}`
+        }
+      },
+      [
+        div({ style: { fontSize: 16 } }, `From Dockstore - ${path}`),
+        div({}, `V. ${version}`),
+        div({
+            style: {
+              display: 'flex', alignItems: 'center',
+              margin: '1rem 0', color: Style.colors.warning
+            }
+          },
+          [
+            icon('warning-standard', { class: 'is-solid', size: 32, style: { marginRight: '0.5rem', flex: '0 0 auto' } }),
+            mutabilityWarning
+          ]),
+        h(Collapse, { title: 'REVIEW WDL' },
+          [h(WDLViewer, { wdl })])
+      ]
+    )
+  }
+
+  renderWorkspaceArea = () => {
+    const { selectedWorkspace, workspaces, importError } = this.state
+
+    return div({},
+      [
+        h(Select, {
+          clearable: false,
+          searchable: true,
+          disabled: !workspaces,
+          placeholder: workspaces ? 'Select a workspace' : 'Loading workspaces...',
+          value: selectedWorkspace,
+          onChange: selectedWorkspace => this.setState({ selectedWorkspace }),
+          options: _.map(workspaces, workspace => {
+            const { name } = workspace.workspace
+            return { value: name, label: name, data: workspace }
+          })
+        }),
+        buttonPrimary(
+          {
+            style: { marginTop: '1rem' },
+            disabled: !selectedWorkspace,
+            onClick: () => this.import()
+          },
+          'Import'),
+        importError ?
+          div({
+            style: { marginTop: '1rem', color: Style.colors.error }
+          }, [
+            icon('error'),
+            JSON.parse(importError).message
+          ]) : null
+      ]
+    )
+  }
+
+  import = () => {
+    const { selectedWorkspace: { data: { workspace: { namespace, name } } } } = this.state
```

Comment:
Having to unpack `data` makes me feel like I'm using the Select wrong.

Reply:
The Select API only requires `value` and `label` in options, where `value` is the underlying data being selected. That would be the workspace, in this case. The `onChange` callback gives you back the whole option, so you could just destructure out `value` at that point, and store just that.